src/erlcloud_cloudformation.erl

```diff
@@ -0,0 +1,412 @@
+            StackEvents = extract_described_stack_events_result(xmerl_xpath:string("/DescribeStackEventsResponse", XmlNode)),
+            {ok, StackEvents, NextToken};
+        {error, Error} ->
+            {error, Error}
+    end.
+
+-spec get_template(string(), aws_config()) -> {ok, cloudformation_list()} | {error, _}.
+get_template(StackName, Config = #aws_config{}) ->
+
+    case cloudformation_request(Config, "GetTemplate", [{"StackName", StackName}]) of
+        {ok, XmlNode} ->
+            extract_template_response(XmlNode);
+        {error, Error} ->
+            {error, Error}
+    end.
+
+-spec get_template_summary(params(), aws_config()) -> {ok, cloudformation_list()} | {error, _}.
+get_template_summary(Params, Config = #aws_config{}) ->
+
+    case cloudformation_request(Config, "GetTemplateSummary", Params) of
+        {ok, XmlNodes} ->
+            extract_template_summary_response(XmlNodes);
+        {error, Error} ->
+            {error, Error}
+    end.
+
+-spec describe_account_limits_all(aws_config()) -> {ok, cloudformation_list()} | {error, _}.
+describe_account_limits_all(Config = #aws_config{}) ->
+    list_all(fun describe_account_limits/2, [], Config, []).
+
+-spec describe_account_limits(aws_config(), string()) -> {ok, cloudformation_list()} | {error, _}.
+describe_account_limits(Params, Config = #aws_config{}) ->
+
+    case cloudformation_request(Config, "DescribeAccountLimits", Params) of
+        {ok, XmlNodes} ->
+            NextToken = erlcloud_xml:get_text("/DescribeAccountLimitsResponse/DescribeAccountLimitsResult/NextToken", XmlNodes, undefined),
+            AccountLimits = extract_accout_limits_response(xmerl_xpath:string("/DescribeAccountLimitsResponse", XmlNodes)),
+            {ok, AccountLimits, NextToken};
+        {error, Error} ->
+            {error, Error}
+    end.
+
+%%==============================================================================
+%% Internal functions
+%%==============================================================================
+
+cloudformation_request(Config = #aws_config{}, Action, ExtraParams) ->
+
+    QParams = [
+        {"Action", Action},
+        {"Version", ?API_VERSION}
+        | ExtraParams],
+
+    erlcloud_aws:aws_request_xml4(post, Config#aws_config.cloudformation_host, "/", QParams, "cloudformation", Config).
+
+list_all(Fun, Params, Config, Acc) ->
+    case Fun(Params, Config) of
+        {ok, Data, NextToken} ->
+            case NextToken of
+                undefined ->
+                    lists:foldl(fun erlang:'++'/2, [], [Data | Acc]);
+                _ ->
+                    list_all(Fun, [{"NextToken", NextToken} | Params], Config, [Data | Acc])
+            end;
+        {error, Reason} ->
+            {error, Reason}
+    end.
+
+extract_stack_summaries(XmlNodes) ->
+    lists:map(fun(T) -> erlcloud_xml:decode([{summaries, "StackSummaries", {optional_map, fun extract_stacks/1}}], T) end, XmlNodes).
```

Comment:
I've just recalled that if something is `undefined` `erlcloud_xml` probably removes that from the result.... Anyway, if so, for top level things I think it is better to have `[]` instead of absense